### PR TITLE
fix(security): private job-dir perms, loopback resolution, agy-path validation

### LINKED ISSUE #35: Hardening: job dir permissions, loopback check, agy path validation
- [ ] Job dirs and files are created 0755/0644 (store.go:81,88,198; the supervisor's out/err via os.Create) and hold prompts and full agy output, which often embed source code. On multi-user hosts with a traversable home those are readable by other users. 0700 dirs and 0600 files throughout is cheap.
- [ ] checkLoopbackAddr accepts the literal "localhost" without resolving it (main.go:125-127); a hosts-file remap defeats the loopback-only intent the comment promises. Resolve and verify all addresses, or document the assumption.
- [ ] AGY_MCP_AGY_PATH is taken verbatim (config.go:34-35) while the PATH-lookup branch fails fast at startup; a typo'd override only surfaces as an exec error on the first job. Stat the override in Resolve for a symmetric startup error.

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

Three security-hardening items (issue #35).

**Private permissions** — job dirs and files hold prompts and full agy output, which often embed source code, and were world/group-readable (`0755`/`0644`) on multi-user hosts. Tighten to owner-only:
- job dirs `0700` (`jobstore.Store.Create`)
- `meta.json` `0600` (`writeMetaAtomic`)
- `exit_code` `0600` (`jobstore.WriteExitCode` and the supervisor's `writeExit`)
- the supervisor's `out`/`err` opened `0600` via `os.OpenFile` instead of `os.Create` (which used `0666`)

Pre-existing dirs from an older binary are not re-tightened by `MkdirAll`, but they age out within one `JobTTL` and new jobs are private immediately.

**Loopback bind check** — `checkLoopbackAddr` accepted the literal `"localhost"` without resolving it, so a hosts-file remap to a routable IP defeated the loopback-only intent. It now resolves a hostname and requires *every* resolved address to be loopback; a literal IP is still checked directly. The resolver is injected (`checkLoopbackAddrResolved`) so the corner cases (remapped localhost, resolve error, empty result, literal-IP fast path) test hermetically without DNS.

**agy path validation** — the `AGY_MCP_AGY_PATH` override was taken verbatim while the PATH-lookup branch failed fast. It is now resolved with `exec.LookPath`, symmetric with that branch: a typo, a non-executable file, or a bad PATH-relative name fails at startup instead of at exec time on the first job, and a bare name still resolves via PATH.

## Related Issues

Closes #35

## Test Plan

- [x] `go test ./... -race`
- [x] `go vet ./...`, `GOOS=darwin go vet ./...`, `GOOS=windows go build/vet ./...`
- [x] `gofmt -l .` clean, `golangci-lint run ./...` 0 issues
- [x] New tests: dir/file perms (in `unix`-tagged files with a cleared umask so they verify the explicit mode, not the ambient umask), loopback hostname resolution (hermetic, injected resolver), and `AGY_MCP_AGY_PATH` missing + non-executable rejection
- [x] Reviewed by a Claude correctness agent and an independent Gemini 3.1 Pro pass; both flagged using `exec.LookPath` over `os.Stat` (adopted)

## Summary by CodeRabbit

* **Bug Fixes**
  * Validate configured agent binary path at startup and normalize to an absolute path.
  * Restrict job data, metadata, output and exit files to owner-only permissions.
  * Tighten loopback bind validation to require all resolved addresses be loopback.

* **Tests**
  * Added and updated tests covering path override validation, permission enforcement, and loopback resolution behavior.

### CODE REVIEW

File: internal/config/config.go [hunk 1]
```diff
@@ -37,7 +37,16 @@ func Resolve() (Config, error) {
...
+		// Resolve the override with LookPath, symmetric with the PATH branch below, so
+		// a typo, a non-executable file, or a bad PATH-relative name fails fast at
+		// startup instead of only at exec time on the first job. LookPath also handles
+		// a bare name (PATH lookup) and locks in an absolute path, shrinking the window
+		// for a PATH change between resolution and exec.
+		resolved, err := exec.LookPath(p)
+		if err != nil {
+			return Config{}, fmt.Errorf("AGY_MCP_AGY_PATH %q: %w", p, err)
+		}
+		c.AgyPath = resolved
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/config/config.go:48]: Good catch, fixed in ad7219d. The supervisor runs agy with cmd.Dir set to the job cwd, so a relative AgyPath would resolve against the wrong directory. I now run filepath.Abs after LookPath (for both the override and PATH branches) and report the pre-Abs value on failure.

File: main.go [hunk 2]
```diff
@@ -115,18 +115,41 @@ func serve() error {
...
+// checkLoopbackAddrResolved is checkLoopbackAddr with the name resolver injected so
+// it is testable without DNS. A literal IP is checked directly; a hostname (such as
+// "localhost") is resolved and EVERY resolved address must be loopback, so a
+// hosts-file remap of "localhost" to a routable IP cannot silently expose the
+// server (the bare-string "localhost" check this replaces trusted the name blindly).
+func checkLoopbackAddrResolved(addr string, lookup func(string) ([]string, error)) error {
 	host, _, err := net.SplitHostPort(addr)
 	if err != nil {
 		host = addr // no port; treat the whole value as the host
 	}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply main.go:136]: Fixed in ad7219d. checkLoopbackAddr now strips the brackets from a portless IPv6 literal like [::1] before net.ParseIP, so it is recognized as loopback instead of falling through to the resolver. The host:port form already worked since SplitHostPort removes the brackets.

File: main.go [hunk 3]
```diff
@@ -115,18 +115,41 @@ func serve() error {
...
-	return fmt.Errorf("http address %q must be loopback only (localhost, 127.0.0.1, or ::1) for HTTP mode", addr)
+	for _, a := range addrs {
+		ip := net.ParseIP(a)
+		if ip == nil || !ip.IsLoopback() {
+			return fmt.Errorf("http host %q resolves to non-loopback address %s; specify a loopback host (e.g. 127.0.0.1:8765) for HTTP mode", host, a)
+		}
+	}
+	return nil
 }
```
[the remark this answers is not in the document]
Comment (tphakala) [reply main.go:167]: I'm going to document this rather than canonicalize the bind address. The TOCTOU is real but not exploitable in the relevant threat model: re-resolving localhost differently requires modifying /etc/hosts or DNS, which needs root, and a root attacker can bypass any loopback restriction anyway (rebind, ptrace, etc.). This check exists to stop an accidental non-loopback bind, not a privileged attacker. Returning a forced canonical IP would also override the user's explicit IPv4-vs-IPv6 choice and could fail on a host without IPv4 loopback. I added a comment on checkLoopbackAddr in ad7219d spelling out the residual TOCTOU and why it's left undefended.
